src/core/MREntity.js

```diff
@@ -52,6 +52,7 @@ export class MREntity extends MRElement {
         });
 
         this.object3D = new THREE.Group();
+        this.object3D.userData.isEntityObject3DRoot = true;
```

Comment:
I introduced `Object3D.userData.isEntityObject3DRoot` to allow to check whether an object is the root object of an entity.

I needed this to traverse the object3Ds only associated with an entity, like

```javascript
class traverse = (obj) => {
  something(obj).
  for (const child of obj.children) {
    if (!obj.userData.isEntityObject3DRoot) {
      traverse(child);
    }
  }
};
traverse(entity.object3D);
```

Honestly I think there might be a more elegant approach. Please suggest better ideas if you come up with.

Reply:
one shorter solution is leaning into the assumption that if object3D is a group, then object3D.children[0] is the root in question

reasoning - when the entities are set up it'll be either the first background object for any mrdiv or the 'object' that should be representing the entity itself given how we instantiate the items in the 'connected' functions - meaning it would actually be the root object

but i'm hesitant to lean towards that as it becomes more overhead long term - and if a user does any of their own javascript modifications or we change the architecture in future, that logic will become a huge edge case

i think what you have, though it's a flag, is straight forward enough for now

in future if/when we can lean into traversing over the entities directly instead of the object3Ds (if that is ever possible) then that reduces our need for the flag, but that's not worth a dive through at the moment.

Reply:
if you're meaning the flag to represent the direct object3d of the entity root, then doing a traversal over the entities instead of the object3d items would be the only alternative, but my one concern would be from the object3ds being overly nested based on the entity hierarchy (you end up with factorially redundant traversals)

so same thing - if it ends up being cleaner great, but the flag approach is straightforward for now